EIPS/eip-4841.md

```diff
@@ -0,0 +1,420 @@
+---
+eip: 4841
+title: Expandable Onchain SVG Images Storage
+description: Expandable SVG images storage model for storing larger-capacity images on Ethereum.
+author: KyungEun Kim <kkekke815@gmail.com>, Soohan Park <725psh@gmail.com>, Soohan Park (@Soohan-Park)
+discussions-to: https://ethereum-magicians.org/t/eip-4841-expandable-onchain-svg-images-storage-structure/8410
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-02-23
+---
+
+## Abstract
+
+This standard proposal is an Expandable Onchain SVG Images Storage on Ethereum that permanently preserves images and prevents tampering, and can store larger-capacity images furthermore.
+
+It is a structure designed to store SVG images with a larger capacity by distributing SVG images in units of tags on Ethereum.
+
+The structure presented by this EIP consists of a total of three layers as shown below.
+
+![StructureDiagram.jpg](../assets/eip-4841/StructureDiagram.jpg)
+
+> **Storage Layer ─** A contract layer that stores distributed SVG images by tags.  
+> **Assemble Layer ─** A contract layer that creates SVG images by combining tags stored in the Storage Layer's contract.  
+> **Property Layer ─** A contract layer that stores the attribute values for which SVG tag to use.
+
+It is designed to flexibly store and utilize larger capacity SVG images by interacting with the above three layer-by-layer contracts each other.
```

Comment:
```suggestion
It is designed to flexibly store and render larger SVG images by interacting with the above three layer-by-layer contracts each other.
```

It might also be helpful to describe how this three layer design saves on fees/storage space.

Reply:
Well.. When we first designed this EIP, it was not designed to save fees/storage space by dividing it into 3 layers.

The reason why we designed with three layers is that when saving SVG images, we designed three layers to store a large capacity and to have flexibility in scalability. (Therefore, using the model increases the cost a little more.)

Should EIP be aimed at saving fees/storage space?

Reply:
> @SamWilsn Well.. When we first designed this EIP, it was not designed to save fees/storage space by dividing it into 3 layers.
> 
> The reason why we designed with three layers is that when saving SVG images, we designed three layers to store a large capacity and to have flexibility in scalability. (Therefore, using the model increases the cost a little more.)
> 
> Should EIP be aimed at saving fees/storage space?

I got an answer from @Pandapip1

Reply:
> It might also be helpful to describe how this three layer design saves on fees/storage space.

thanks for your feedback :)  
well I think it will be placed more natural in Rationale. How do you think about that?